src/ucp/core/ucp_worker.c

```diff
@@ -484,6 +484,29 @@ static ucp_ep_h ucp_worker_find_lane(ucs_list_link_t *ep_list, uct_ep_h uct_ep,
     return NULL;
 }
 
+/**
+ * FLUSH_CANCEL operation might be on pending queue due to
+ * UCS_ERR_NO_RESOURCES, so need to purge the queue to resubmit the
+ * operation. We need to resubmit the FLUSH_CANCEL operation on the same
+ * failed lane, in order to make sure all previous outstanding
+ * operations are completed before destroying the failed endpoint.
+ */
+static void ucp_discard_lane_ff(uct_ep_h uct_ep)
+{
+    ucs_queue_head_t tmp_q;
+    uct_pending_req_t *uct_req;
+    ucp_request_t *ucp_req;
+
+    /* @ref ucs_arbiter_t does not support recursive calls, so use temporary
+     * queue */
+    ucs_queue_head_init(&tmp_q);
+    uct_ep_pending_purge(uct_ep, ucp_request_purge_enqueue_cb, &tmp_q);
+    ucs_queue_for_each_extract(uct_req, &tmp_q, priv, 1) {
+        ucp_req = ucs_container_of(uct_req, ucp_request_t, send.uct);
```

Comment:
no need to define uct_req:
`ucs_queue_for_each_extract(ucp_req, &tmp_q, send.uct.priv, 1) {`